Zend/zend_compile.c

```diff
@@ -6165,6 +6169,7 @@ void zend_compile_func_decl(znode *result, zend_ast *ast, zend_bool toplevel) /*
 	zend_ast *stmt_ast = decl->child[2];
 	zend_ast *return_type_ast = decl->child[3];
 	zend_bool is_method = decl->kind == ZEND_AST_METHOD;
+	zend_bool is___tostring = is_method && CG(active_class_entry)->__tostring == NULL;
```

Comment:
I think I've found a bug in your code patch. This should be `!= NULL`, unless I've missunderstood something. I see later checks doing `!= NULL` to assert is__tostring

Reply:
If this is indeed a bug, then you need to figure out how your .phpt tests never caught this, and design a .phpt test specifically for this code path.

Reply:
No bug, that's on purpose: we suppose this is the __toString method when it's not already defined, then we revert to false L6206 if it's not.

Reply:
I'm glad to hear this and I suspected as much too, but I thought it was worth making you double check it.